spicy/toolchain/src/compiler/visitors/validator.cc

```diff
@@ -579,6 +580,22 @@ struct VisitorPost : public hilti::visitor::PreOrder<void, VisitorPost>, public
                 if ( a.tag() == "&size" || a.tag() == "&max-size" ) {
                     if ( ! a.hasValue() )
                         error(fmt("%s must provide an expression", a.tag()), p);
+                    else {
+                        auto v = hilti::visitor::PreOrder<>();
+                        for ( auto i : v.walk(a.value()) ) {
+                            auto&& member = i.node.tryAs<operator_::unit::MemberNonConst>();
+                            if ( ! member || ! member->hasOp0() )
+                                continue;
+
+                            auto&& self = member->op0().tryAs<hilti::expression::ResolvedID>();
+                            if ( ! self || self->id().str() != "self" )
+                                continue;
+
+                            error(fmt("%s expression cannot depend on information available only after parsing of unit",
```

Comment:
I'd make this more explicit: "... cannot use `self` because that depends on ...".